### PR TITLE
SONAR-8335 log property per jvm

### PR DESCRIPTION
**Pull request by sns-seb**

### CODE REVIEW

File: server/sonar-process/src/main/java/org/sonar/process/LogbackHelper.java [hunk 1]
```diff
@@ -73,23 +88,216 @@ public LoggerContextListener enableJulChangePropagation(LoggerContext loggerCont
...
+   * the max number of files defined in property {@link #MAX_FILES_PROPERTY}</li>
+   * <li>the logs will follow the specified log pattern</li>
+   * </ul>
+   * </p>
+   *
+   * @see #buildLogPattern(RootLoggerConfig)
+   */
+  public FileAppender<ILoggingEvent> configureGlobalFileLog(Props props, RootLoggerConfig config, String logPattern) {
+    LoggerContext ctx = getRootContext();
+    Logger rootLogger = ctx.getLogger(ROOT_LOGGER_NAME);
```
Comment (SonarTech) [inline server/sonar-process/src/main/java/org/sonar/process/LogbackHelper.java]: MAJOR Make the "rootLogger" logger private static final and rename it to comply with the format "LOG(?:GER)?". [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1312)

File: server/sonar-process/src/main/java/org/sonar/process/LogbackHelper.java [hunk 2]
```diff
@@ -73,23 +88,216 @@ public LoggerContextListener enableJulChangePropagation(LoggerContext loggerCont
...
+    return configureRootLogLevel(newLevel);
+  }
+
+  /**
+   * Configure the log level of the root logger to the specified level.
+   *
+   * @throws IllegalArgumentException if the specified level is not one of {@link #ALLOWED_ROOT_LOG_LEVELS}
+   */
+  public Level configureRootLogLevel(Level newLevel) {
+    Logger rootLogger = getRootContext().getLogger(ROOT_LOGGER_NAME);
```
Comment (SonarTech) [inline server/sonar-process/src/main/java/org/sonar/process/LogbackHelper.java]: MAJOR Make the "rootLogger" logger private static final and rename it to comply with the format "LOG(?:GER)?". [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1312)

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues
* MAJOR 2 major

Watch the comments in this conversation to review them.

**Comment by sns-seb** [comment]
@simonbrandhof this PR is based on #1370 , review only the last two commits

File: sonar-application/src/main/assembly/conf/sonar.properties [hunk 3]
```diff
@@ -213,9 +213,17 @@
 #--------------------------------------------------------------------------------------------------
 # LOGGING
 
-# Level of logs. Supported values are INFO(default), DEBUG and TRACE (DEBUG + SQL + ES requests)
+# Global level of logs (applies to App, Web and CE processes).
+# Supported values are INFO (default), DEBUG and TRACE (on web server: TRACE = DEBUG + SQL + ES requests)
```
Comment (simonbrandhof) [inline sonar-application/src/main/assembly/conf/sonar.properties]: TRACE level should not have a specific behaviour in web server. It should enable all TRACE logs, not only SQL and ES.

File: server/sonar-server/src/main/java/org/sonar/ce/log/CeProcessLogging.java [hunk 4]
```diff
@@ -19,16 +19,19 @@
...
 
+import org.sonar.process.ProcessId;
 import org.sonar.process.Props;
 import org.sonar.server.app.ServerProcessLogging;
 
+import static org.sonar.ce.log.CeLogging.MDC_CE_TASK_UUID;
+
 /**
- * Configure logback for the Compute Engine process. Logs are written to file "ce.log" in SQ's log directory.
+ * Configure logback for the Conpute Engine process. Logs are written to file "ce.log" in SQ's log directory.
```
Comment (simonbrandhof) [inline server/sonar-server/src/main/java/org/sonar/ce/log/CeProcessLogging.java]: Typo in "Conpute".

File: sonar-application/src/main/assembly/conf/sonar.properties (hunk 3 above)
Comment (sns-seb) [inline sonar-application/src/main/assembly/conf/sonar.properties]: I only rephrased what was already there. I'll just drop it.

File: server/sonar-process/src/main/java/org/sonar/process/LogbackHelper.java [hunk 5]
```diff
@@ -73,23 +92,243 @@ public LoggerContextListener enableJulChangePropagation(LoggerContext loggerCont
...
+   * Configure the log level of the root logger reading the value of specified properties. The lowest level will be
+   * applied.
+   *
+   * @throws IllegalArgumentException if the value of the specified property is not one of {@link #ALLOWED_ROOT_LOG_LEVELS}
+   */
+  public Level configureRootLogLevel(Props props, String... propertyKeys) {
+    return configureLoggerLogLevel(getRootContext().getLogger(ROOT_LOGGER_NAME), props, propertyKeys);
+  }
+
+  private Level configureLoggerLogLevel(Logger logger, Props props, String... propertyKeys) {
```
Comment (SonarTech) [inline server/sonar-process/src/main/java/org/sonar/process/LogbackHelper.java]: MAJOR Make "configureLoggerLogLevel" a "static" method. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2325)

**Comment by SonarTech** [comment]
SonarQube analysis reported 4 issues
* MAJOR 4 major

Watch the comments in this conversation to review them.

File: server/sonar-process/src/main/java/org/sonar/process/LogbackHelper.java [hunk 6]
```diff
@@ -275,13 +312,17 @@ public Level configureRootLogLevel(Props props, String propertyKey) {
    */
   public Level configureRootLogLevel(Level newLevel) {
     Logger rootLogger = getRootContext().getLogger(ROOT_LOGGER_NAME);
```
Comment (SonarTech) [inline server/sonar-process/src/main/java/org/sonar/process/LogbackHelper.java]: MAJOR Make the "rootLogger" logger private static final and rename it to comply with the format "LOG(?:GER)?". [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1312)

**Comment by SonarTech** [comment]
SonarQube analysis reported 4 issues
* MAJOR 4 major

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [LogbackHelper.java#L333](https://github.com/SonarSource/sonarqube/blob/50684ff5ab0349865af38eb1d0892e7e8e521d06/server/sonar-process/src/main/java/org/sonar/process/LogbackHelper.java#L333): Make the "logger" logger private static final and rename it to comply with the format "LOG(?:GER)?". [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1312)

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues
* MAJOR 2 major

Watch the comments in this conversation to review them.

**Pull request merged**